lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -0,0 +1,379 @@
+        # other exceptions
+        except Exception as e:
+            module.fail_json(changed=False, msg="Failed to Delete directory into Vm VMware exception:%s" % e)
+
+    return True
+
+
+def fetch(module, content, vm):
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+
+    if "src" not in module.params["fetch"]:
+        module.fail_json(msg="fetch.src is mandatory")
+    if "dest" not in module.params["fetch"]:
+        module.fail_json(msg="fetch.dest is mandatory")
+
+    dest = module.params["fetch"]['dest']
+    src = module.params['fetch']['src']
+
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_manager = content.guestOperationsManager.fileManager
+
+    try:
+        fileTransferInfo = file_manager.InitiateFileTransferFromGuest(vm=vm, auth=creds,
+                                                                      guestFilePath=src)
+    except vim.fault.FileFault as e:
+        module.fail_json(msg="FileFault:%s" % e.msg)
+    except vim.fault.GuestPermissionDenied:
+        module.fail_json(msg="Permission denied to fetch file %s" % src)
+    except vim.fault.InvalidGuestLogin:
+        module.fail_json(msg="Invalid guest login for user %s" % vm_username)
+    # other exceptions
+    except Exception as e:
+        module.fail_json(changed=False, msg="Failed to Fetch file from Vm VMware exception:%s" % e)
+
+    url = fileTransferInfo.url
+    resp, info = urls.fetch_url(module, url, method="GET")
+    with open(dest, "wb") as local_file:
+        local_file.write(resp.read())
+
+    return True
+
+
+def copy(module, content, vm):
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+
+    if "src" not in module.params["copy"]:
+        module.fail_json(msg="copy.src is mandatory")
+    if "dest" not in module.params["copy"]:
+        module.fail_json(msg="copy.dest is mandatory")
+    if "overwrite" in module.params["copy"]:
+        overwrite = module.params["copy"]["overwrite"]
+    else:
+        overwrite = False
+
+    dest = module.params["copy"]['dest']
+    src = module.params['copy']['src']
+    b_src = to_bytes(src, errors='surrogate_or_strict')
+    if not os.path.exists(b_src):
+        module.fail_json(msg="Source %s not found" % (src))
+    if not os.access(b_src, os.R_OK):
+        module.fail_json(msg="Source %s not readable" % (src))
+    if os.path.isdir(b_src):
+        module.fail_json(msg="copy does not support copy of directory: %s" % (src))
+
+    data = None
+    with open(b_src, "r") as local_file:
+        data = local_file.read()
+    file_size = os.path.getsize(b_src)
+
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_attributes = vim.vm.guest.FileManager.FileAttributes()
+    file_manager = content.guestOperationsManager.fileManager
+    try:
```

Comment:
Should there be a check if the file exists if overwrite is `False`, and if it does, return `changed=False`?